tools/python/openlr/quality.py

```diff
@@ -143,32 +149,38 @@ def parse_segments(tree, limit):
         # within limit are considered accurate, so golden path should be equal
         # matched path.
         golden_route = parse_route(s.find('GoldenRoute')) or matched_route
-        if not matched_route and not golden_route:
-            raise ValueError('At least one of golden route or matched route should be present')
+        yield Segment(segment_id, golden_route, matched_route)
+
+def calculate(tree):
+    result = {}
+    for s in parse_segments(tree, args.limit):
         try:
-            yield Segment(segment_id, matched_route, golden_route)
+            result[s.segment_id] = common_part(s.golden_route, s.matched_route)
         except:
-            print('Broken segment is {}'.format(segment_id))
+            print('Something is wrong with segment {}'.format(s))
             raise
-
-def calculate(tree):
-    return {
-        s.segment_id: common_part(s.golden_route, s.matched_route)
-        for s in parse_segments(tree, args.limit)
-    }
+    return result
 
 def merge(src, dst):
-    golden_routes = {
-        int(s.find('.//ReportSegmentID').text) : s.find('GoldenRoute')
-        for s in src.findall('Segment')
-    }
+    golden_routes = dict(
+        filter(
```

Comment:
I think check for emptiness later is more readable than this filter

Reply:
Done.